core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -674,6 +677,22 @@ private[spark] class Executor(
     }
   }
 
+  private def setMDCForTask(taskDescription: TaskDescription): Unit = {
+    val properties = taskDescription.properties
+
+    def getProperty(id: String): String = {
+      Option(properties.getProperty(id)).getOrElse(conf.get(id, ""))
+    }
+
+    org.slf4j.MDC.put("appId", getProperty("spark.app.id"))
+    org.slf4j.MDC.put("appName", getProperty("spark.app.name"))
```

Comment:
These are totally useless codes. If you really tested locally with default setting, you'll find the appId/appName is always empty.

Reply:
so what you propose, how to do that in tests?

Reply:
But when debugging I see that in conf it presently, so why it not being propagated to the task?

Reply:
> But when debugging I see that in conf it presently, so why it not being propagated to the task?

You may not run with default settings.

Reply:
> so what you propose, how to do that in tests?

I always propose to use task name. However, I also realized that we're facing different use cases and want to put different values. Actually, I think that it's not really make sense to put the same value for different task threads using MDC. AFAIK, MDC is more appropriate for the scenario where we want to distinguish logs from concurrent threads/request/tasks. So, it's more likely to put a unique value(e.g. task name) here. As for your case, I'd recommend you to use custom way to add appId/appName without polluting task threads by default.

Reply:
What do you mean default setting? I did not change anything in tests/resources.

Reply:
> What do you mean default setting? I did not change anything in tests/resources.

Which Spark version do you use?

Reply:
master